src/main/java/org/kitteh/irc/client/library/implementation/NettyManager.java

```diff
@@ -286,18 +305,112 @@ public String toString() {
         }
     }
 
+    static class DCCConnection extends ChannelInitializer<SocketChannel> {
+        private final IRCDCCExchange exchange;
+        private final InternalClient client;
+        private int connectionsMade;
+
+        private DCCConnection(IRCDCCExchange ex, InternalClient client) {
+            this.exchange = ex;
+            this.client = client;
+        }
+
+        @Override
+        public void initChannel(SocketChannel channel) throws Exception {
+            if (this.connectionsMade > 0) {
+                // Only one connection is allowed.
+                channel.close();
+                return;
+            }
+            this.connectionsMade++;
+            this.exchange.setNettyChannel(channel);
+            dccConnections.computeIfAbsent(this.client, c -> new ArrayList<>()).add(channel);
+
+            // Outbound - Processed in pipeline back to front.
+            channel.pipeline().addFirst("[OUTPUT] Output listener", new MessageToMessageEncoder<String>() {
```

Comment:
This looks like lots of duplication from above - it should be possible to share most, if not all, of it.